jaxb-ri/txw/compiler/src/main/java/com/sun/tools/txw2/TxwOptions.java

```diff
@@ -58,4 +68,145 @@
      * {@link XmlNamespace} annotation.
      */
     public boolean noPackageNamespace;
+
+    public Language language;
+
+    public enum Language {
+        XML,
+        XMLSCHEMA,
+        COMPACT
+    }
+
+    void parseArguments(String[] args) throws BadCommandLineException {
+        String src = null;
+        for (int i = 0; i < args.length; i++) {
+            if (args[i].length() == 0) {
+                throw new BadCommandLineException();
+            }
+            if (args[i].charAt(0) == '-') {
+                int j = parseArgument(args, i);
+                if (j == 0) {
+                    throw new BadCommandLineException(getMessage("unrecognizedParameter", args[i]));
+                }
+                i += (j - 1);
+            } else {
+                src = args[i];
+            }
+        }
+        if (_package == null) {
+            _package = codeModel.rootPackage();
+        }
+        if (codeWriter == null) {
+            codeWriter = new SingleStreamCodeWriter(System.out);
+        }
+        if (src == null) {
+            throw new BadCommandLineException(getMessage("missingOperand"));
+        }
+        try {
+            source = makeSourceSchema(src, errorListener);
+        } catch (MalformedURLException mue) {
+            throw new BadCommandLineException(mue.getMessage(), mue);
+        }
+    }
+
+    private int parseArgument(String[] args, int i) throws BadCommandLineException {
+        if (args[i].equals("-o")) {
```

Comment:
what about using switch(args[i].equals("-o")) { ...

Reply:
do you want me to go through all the codebase of jaxb-ri and jaxws to change this (>50 occurrences)?

Reply:
No, just in your diff where you are already touching the code.

Reply:
But OK, let's keep it as is.